### ISSUE TITLE
Cloudformation Nested Stacks should allow parameters from parent

### ISSUE THREAD

**Issue by ashleyfrieze** [localstack/localstack#2111]
I spent a couple of hours yesterday trying to deploy a parent stack with two nested dynamodb table stacks. Both stacks were to be built from the same template and so provided a TableName parameter for the general purpose child stack to use along with its standard settings. This is one of the primary use cases for child stacks.

It didn’t work, and looking at #1921 it appears that the code treats the child stack as though it should just run independently.

I suspect that #2079 needs the obverse of this, which is the collection of outputs from a child to use with a peer.

**Comment by ashleyfrieze** [comment]
Here's some extra info.

Parent file:

```yml
AWSTemplateFormatVersion: 2010-09-09
Description: SomeParent

Parameters:
  GlobalAppStageName:
    Description: Global - App Stage Name
    Type: String
    Default: 'dev'

Resources:
  EngineTable:
    Type: AWS::CloudFormation::Stack
    Properties:
      TemplateURL: ./dynamo-component.yml
      Parameters:
        GlobalAppStageName: !Ref GlobalAppStageName
        TypeName: 'foo'
  FleetTable:
    Type: AWS::CloudFormation::Stack
    Properties:
      TemplateURL: ./dynamo-component.yml
      Parameters:
        GlobalAppStageName: !Ref GlobalAppStageName
        TypeName: 'bar'
```

**dynamo-component.yml**

```yml
AWSTemplateFormatVersion: 2010-09-09

Parameters:
  GlobalAppStageName:
    Description: Global - App Stage Name
    Type: String
    Default: 'local'
  TypeName:
    Description: Name of table
    Type: String

Resources:
  EngineTable:
    Type: AWS::DynamoDB::Table
    Properties:
      AttributeDefinitions:
        -
          AttributeName: "id"
          AttributeType: "S"
      BillingMode: PROVISIONED
      KeySchema:
        -
          AttributeName: "id"
          KeyType: "HASH"
      ProvisionedThroughput:
        ReadCapacityUnits: "5"
        WriteCapacityUnits: "5"
      TableName: !Sub 'my-foo-prefix-${GlobalAppStageName}-${TypeName}'
```

As with #1921 the child stack can be made individually, but rolling it into a parent stack doesn't work. Note, how both the stage and the table name are parameters that need to be passed down to the child here.

**Comment by ashleyfrieze** [comment]
@whummer Thanks!

**Comment by whummer** [comment]
Thanks for reporting @ashleyfrieze . This should be fixed in #2153 (available in `latest` Docker image) - please report here if the problem persists. Thanks
